cmd/cloudshell_open/main.go

```diff
@@ -250,48 +250,58 @@ func run(opts runOpts) error {
 		return err
 	}
 
-	exists, err := dockerFileExists(appDir)
-	jibMaven := false
-	if err != nil {
-		return err
+	projectEnv := fmt.Sprintf("GOOGLE_CLOUD_PROJECT=%s", project)
+	regionEnv := fmt.Sprintf("GOOGLE_CLOUD_REGION=%s", region)
+	serviceEnv := fmt.Sprintf("K_SERVICE=%s", serviceName)
+	imageEnv := fmt.Sprintf("IMAGE_URL=%s", image)
+	appDirEnv := fmt.Sprintf("APP_DIR=%s", appDir)
+	pathEnv := fmt.Sprintf("PATH=%s", os.Getenv("PATH"))
+
+	hookEnvs := append([]string{projectEnv, regionEnv, serviceEnv, imageEnv, appDirEnv, pathEnv}, envs...)
+
+	pushImage := true
+
+	if appFile.Hooks.PreBuild.Commands != nil {
+		err = runScripts(appDir, appFile.Hooks.PreBuild.Commands, hookEnvs)
 	}
-	if exists {
+
+	skipBuild := appFile.Build.Skip != nil && *appFile.Build.Skip == true
+
+	if skipBuild {
+		// do nothing
```

Comment:
do you think we should print an "info" or something? would this confuse end-users a lot?